pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -390,6 +424,67 @@ func handleZedrouterRestarted(ctxArg interface{}, restartCounter int) {
 	}
 }
 
+// handleAppInstanceStatusCreate - Handle AIS create. Publish AppStatusSummary to ledmanager
+func handleAppInstanceStatusCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	ctx := ctxArg.(*zedmanagerContext)
+	publishAppInstanceSummary(ctx)
+}
+
+// handleAppInstanceStatusModify - Handle AIS modify. Publish AppStatusSummary to ledmanager
+func handleAppInstanceStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}, oldStatusArg interface{}) {
+	ctx := ctxArg.(*zedmanagerContext)
+	publishAppInstanceSummary(ctx)
+}
+
+// handleAppInstanceStatusDelete - Handle AIS delete. Publish AppStatusSummary to ledmanager
+func handleAppInstanceStatusDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	ctx := ctxArg.(*zedmanagerContext)
+	publishAppInstanceSummary(ctx)
+}
+
+func publishAppInstanceSummary(ctxPtr *zedmanagerContext) {
+
+	summary := types.AppInstanceSummary{
+		TotalStarting: 0,
+		TotalRunning:  0,
+		TotalStopping: 0,
+		TotalError:    0,
+	}
+	items := ctxPtr.pubAppInstanceStatus.GetAll()
+	for _, st := range items {
+		status := st.(types.AppInstanceStatus)
+
+		if status.EffectiveActivate && status.Activated {
+			summary.TotalRunning++
+		} else {
+
+			switch status.State {
+			case types.INITIAL, types.DOWNLOADING, types.LOADING, types.CREATING_VOLUME, types.BOOTING:
+				summary.TotalStarting++
+			case types.HALTING:
+				summary.TotalStopping++
+			default:
+
+				if len(status.Error) > 0 {
```

Comment:
Note that this doesn't catch the case when it is in e.g., Downloading state and has an error.

Reply:
Ok where can I look for those errors ?